packages/pytest-simcore/src/pytest_simcore/helpers/httpx_calls_capture_openapi.py

```diff
@@ -0,0 +1,122 @@
+from pathlib import Path
+from typing import Any, Literal, get_args
+from urllib.parse import unquote
+
+import httpx
+import jsonref
+from pydantic import parse_obj_as
+from settings_library.catalog import CatalogSettings
+from settings_library.director_v2 import DirectorV2Settings
+from settings_library.storage import StorageSettings
+from settings_library.webserver import WebServerSettings
+
+from .httpx_calls_capture_errors import (
+    OpenApiSpecError,
+    PathNotInOpenApiSpecError,
+    VerbNotInPathError,
+)
+from .httpx_calls_capture_parameters import (
+    CapturedParameter,
+    CapturedParameterSchema,
+    PathDescription,
+)
+
+service_hosts = Literal["storage", "catalog", "webserver", "director-v2"]
+
+assert CapturedParameterSchema  # nosec
+
+
+def _get_openapi_specs(host: service_hosts) -> dict[str, Any]:
+    url: str
+    if host == "storage":
+        settings = StorageSettings.create_from_envs()
+        url = settings.base_url + "/dev/doc/swagger.json"
+    elif host == "catalog":
+        settings = CatalogSettings.create_from_envs()
+        url = settings.base_url + "/api/v0/openapi.json"
+    elif host == "webserver":
+        settings = WebServerSettings.create_from_envs()
+        url = settings.base_url + "/dev/doc/swagger.json"
+    elif host == "director-v2":
+        settings = DirectorV2Settings.create_from_envs()
+        url = settings.base_url + "/api/v2/openapi.json"
+    else:
+        msg = f"{host=} has not been added yet to the testing system. Please do so yourself"
+        raise OpenApiSpecError(msg)
+    with httpx.Client() as session:
+        # http://127.0.0.1:30010/dev/doc/swagger.json
+        # http://127.0.0.1:8006/api/v0/openapi.json
```

Comment:
Perhaps delete these two comments. I added them for myself in order to know the path of the openapi.json depending on which type of services it is.